rust/noosphere-ns/src/name_system.rs

```diff
@@ -49,11 +50,11 @@ where
     pub(crate) key_material: K,
     pub(crate) store: SphereDb<S>,
     /// Map of sphere DIDs to [NSRecord] hosted/propagated by this name system.
-    hosted_records: HashMap<String, NSRecord>,
+    hosted_records: Mutex<HashMap<Did, NSRecord>>,
```

Comment:
This map isn't fully used (as auto-propagation isn't persistent), need to determine if persistence in libp2p is sufficient when backing via some store, where we do not need to manage "hosted records" in the client. #99